src/runtime/slot-composer.ts

```diff
@@ -8,34 +8,23 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-import {assert} from '../platform/assert-web.js';
 import {Arc} from './arc.js';
-import {Description} from './description.js';
-import {ModalityHandler} from './modality-handler.js';
 import {Modality} from './modality.js';
 import {Particle} from './recipe/particle.js';
-import {SlotConsumer} from './slot-consumer.js';
-import {ProvidedSlotContext, SlotContext} from './slot-context.js';
+import {ProvideSlotConnectionSpec} from './particle-spec.js';
 import {logsFactory} from '../platform/logs-factory.js';
 
 const {log, warn} = logsFactory('SlotComposer', 'brown');
 
 export type SlotComposerOptions = {
   modalityName?: string;
-  modalityHandler?: ModalityHandler;
   noRoot?: boolean;
-  rootContainer?;
-  rootContext?;
-  containerKind?: string;
-  containers?;
 };
 
 export class SlotComposer {
-  private readonly _containerKind: string;
   readonly modality: Modality;
-  readonly modalityHandler: ModalityHandler;
-  private readonly _consumers: SlotConsumer[] = [];
-  protected _contexts: SlotContext[] = [];
+  protected _contexts = [];
+  arc?;
```

Comment:
Don't we need a type here? Does this actually parse/work, and it just is inferred? Does it need to be public, and optional?

Reply:
I left this unfinished because I would like to have a stronger mental-model of the information architecture. The back-reference to the arc is used for exactly one thing: for channeling events back from UiBroker through SlotComposer into the Arc. 
IMO, the relationship of SlotComposer, the PEC (which is actually a _PEH_, btw), and the Arc is unstructured in the current code. 
https://github.com/PolymerLabs/arcs/issues/4563